File: src/py_listener/py_listener/listener.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class Listener(Node):

    def __init__(self):
        super().__init__('listener')
        self.subscription = self.create_subscription(String,'topic',self.call,10)
        self.subscription

    def call(self, msg):
        self.get_logger().info(f'shiko heard: "{msg.data}"')

def main(args=None):
    rclpy.init(args=args)
    l = Listener()
    rclpy.spin(l)
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: src/c_listener/src/listener.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

class Listener : public rclcpp::Node
{
public:
  Listener() : Node("listener")
  {
    subscription_ = this->create_subscription<std_msgs::msg::String>(
      "topic", 10, std::bind(&Listener::call, this, std::placeholders::_1));
  }

private:
  void call(const std_msgs::msg::String::SharedPtr msg) const
  {
    RCLCPP_INFO(this->get_logger(), "shiko heard: '%s'", msg->data.c_str());
  }

  rclcpp::Subscription<std_msgs::msg::String>::SharedPtr subscription_;
};

int main(int argc, char *argv[])
{
  rclcpp::init(argc, argv);
  auto node = std::make_shared<Listener>();
  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}
